typescript-defs/wot.d.ts

```diff
@@ -0,0 +1,182 @@
+//this NOT part of the spec, but needed to polyfill Observable into TypeScript 
+import { Observable } from 'rxjs'
+
+export as namespace wot;
+
+export default wot;
+
+declare let wot: WoTFactory;
+
+interface WoTFactory {
+    discover(filter?: Object): Observable<ConsumedThing>;
+    retrieve(url: string): Promise<ConsumedThing>;
+    createLocalThing(init: ThingInit): Promise<ConsumedThing>;
+}
+
+interface ThingInit {
+    name?: string;
+    url?: string;
+    description?: object;
+}
+
+declare enum DiscoveryType {
+    "any",
+    "local",
+    "nearby",
+    "directory",
+    "broadcast",
+    "other"
+}
+
+interface ThingFilter extends ThingInit {
+    type: string | DiscoveryType
+}
+
+interface ConsumedThing {
+    /** name of the Thing */
+    readonly name: string
+
+    /** invokes an action on the target thing 
+     * @param actionName Name of the action to invoke
+     * @param parameter optional json object to supply parameters  
+    */
+    invokeAction(actionName: string, parameter?: any): Promise<any>
+
+    /**
+     * Set a given property
+     * @param propertyName Name of the property
+     * @param newValue value to be set  
+     */
+    setProperty(propertyName: string, newValue: any): Promise<void>
+
+    /**
+     * Read a given property
+     * @param propertyName Name of the property 
+     */
+    getProperty(propertyName: string): Promise<any>
+
+    addListener(eventName: string, listener: (event: Event) => void): ConsumedThing
+    removeListener(eventName: string, listener: (event: Event) => void): ConsumedThing
+    removeAllListeners(eventName: string): ConsumedThing
+
+    /**
+     * Retrive the thing description for this object
+     */
+    getDescription(): Object
+}
+
+interface PropertyChangeEvent extends Event {
+    readonly data : {
+        name : string;
+        value : any;
+        oldValue: any;
+    }
+}
+
+interface ActionInvocationEvent extends Event {
+    readonly data : {
+        name : string;
+        returnValue: any;
+    }
+}
+
+interface ThingDescriptionChangeEvent extends Event {
+    readonly data : {
+        name : string;
+        data: any; // : ThingPropertyInit | ThingActionInit | ThingEventInit 
+        type: TDChangeType;
```

Comment:
this was actually a leftover from editing it, the comment was meant to replace the any as soon as the types were defined